src/v2/stores/socket.js

```diff
@@ -1,22 +1,37 @@
+import {observable, decorate, action} from 'mobx';
 import {initSocket} from 'v2/api/socket';
 import networkOverview from 'v2/stores/networkOverview';
 import nodes from 'v2/stores/nodes';
 
+import * as EndpointConfig from '../../EndpointConfig';
+
 const socketActions = {
   actions: {
     'global-info': networkOverview.updateGlobalStats,
     blk: networkOverview.addBlock,
     'cluster-info': nodes.updateClusterInfo,
     'txns-by-prgid': networkOverview.addTransaction,
   },
+  endpointName: EndpointConfig.getEndpointName(),
   onMessage(event) {
     const {t, m} = JSON.parse(event.data);
     this.actions[t](m);
   },
   init() {
+    if (this.ws) {
+      this.ws.close();
```

Comment:
nit: set this.ws back to `null` after the `close()` as `initSocket()` can throw

Reply:
@mvines done